web/client-api/src/main/java/io/deephaven/web/client/api/parse/JsDataHandler.java

```diff
@@ -0,0 +1,488 @@
+                    }
+                }
+
+                if (boolValue != FALSE_BOOLEAN_AS_BYTE && boolValue != TRUE_BOOLEAN_AS_BYTE
+                        && boolValue != NULL_BOOLEAN_AS_BYTE) {
+                    throw new IllegalArgumentException("Can't handle " + val + " as a boolean value");
+                }
+
+                // write the value, and mark non-null if necessary
+                if (boolValue != NULL_BOOLEAN_AS_BYTE) {
+                    nulls.set(i);
+                }
+                payload.setAt(i, (double) boolValue);
+            }
+
+            // validity, then payload
+            addBuffer.apply(makeValidityBuffer(nullCount, nulls));
+            addBuffer.apply(new Uint8Array(Js.<TypedArray>uncheckedCast(payload).buffer));
+
+            addNode.apply(new Node(data.length, nullCount));
+        }
+    },
+    BIG_DECIMAL(Type.Binary, "java.util.BigDecimal") {
+        @Override
+        public double writeType(Builder builder) {
+            return Binary.createBinary(builder);
+        }
+    },
+    BIG_INTEGER(Type.Binary, "java.util.BigInteger") {
+        @Override
+        public double writeType(Builder builder) {
+            return Binary.createBinary(builder);
+        }
+    },
+    LOCAL_DATE(Type.FixedSizeBinary, "java.time.LocalDate") {
+        @Override
+        public double writeType(Builder builder) {
+            return FixedSizeBinary.createFixedSizeBinary(builder, 6);
+        }
+    },
+    LOCAL_TIME(Type.FixedSizeBinary, "java.time.LocalTime") {
+        @Override
+        public double writeType(Builder builder) {
+            return FixedSizeBinary.createFixedSizeBinary(builder, 7);
+        }
+    },
+    // LIST(),
+    ;
+
+    public static JsDataHandler getHandler(String deephavenType) {
+        return Arrays
+                .stream(values())
+                .filter(h -> h.deephavenType().equals(deephavenType))
+                .findFirst()
+                .orElseThrow(
+                        () -> new IllegalArgumentException("No support for handling data of type " + deephavenType));
+    }
+
+    /**
+     * Helper to write numeric types that JS can represent in a consistent way.
+     *
+     * @param data the data passed from the user
+     * @param addNode a callback to append a node to the arrow payload
+     * @param addBuffer a callback to append a buffer to the arrow payload
+     * @param bytesPerElement the width of the arrow/deephaven type being created
+     * @param nullValue the value to write in case of null data
+     * @param bufferConstructor a constructor to produce a typedarray for the data being created
+     */
+    private static void writeSimpleNumbers(Object[] data, JsConsumer<Node> addNode, JsConsumer<Uint8Array> addBuffer,
+            double bytesPerElement, double nullValue, JsFunction<ArrayBuffer, ? extends TypedArray> bufferConstructor) {
+        int nullCount = 0;
+        BitSet nulls = new BitSet(data.length);
```

Comment:
same comment here on the null counting; I'd send the empty validity buffer explicitly